### PR TITLE
feat: Add helper methods in S3 and Athena to get table and files

### PR DESCRIPTION
**Pull request by arihantsurana**
### Link to Github Issue
---
* NA

### What changes are proposed in this PR?
---
* Add feature in AthenaUtil to retrive table's location on s3
* Add feature in S3Util to list all files under a given location

### How was this tested?
---
* Unittests

### CODE REVIEW

File: hip_data_tools/aws/athena.py [hunk 1]
```diff
@@ -110,12 +110,16 @@ def _show_result(self, execution_id, max_result_size=1000):
             [[val['VarCharValue'] for val in row['Data']] for row in results['ResultSet']['Rows']])
 
     def _get_query_result(self, execution_id, max_result_size=1000):
-        athena = self.conn.client(self.boto_type)
+        athena = self._client()
         results = athena.get_query_results(QueryExecutionId=execution_id,
                                            MaxResults=max_result_size)
         # TODO: Add ability to parse pages larger than 1000 rows
```
Comment (datascienceChris) [inline hip_data_tools/aws/athena.py]: are you keeping TODOs in line or at the top of the file?

File: hip_data_tools/aws/athena.py [hunk 2]
```diff
@@ -110,12 +110,16 @@ def _show_result(self, execution_id, max_result_size=1000):
...
         return results
 
+    def _client(self):
+        athena = self.conn.client(self.boto_type)
```
Comment (datascienceChris) [inline hip_data_tools/aws/athena.py]: do you want to instantiate a client everytime this method is called? Should it not be called _get_client? That would closer to the functionality

File: hip_data_tools/aws/athena.py [hunk 3]
```diff
@@ -212,6 +216,24 @@ def get_table_ddl(self, table):
             ddl = ddl + "\n"
         return ddl
 
+    def get_table_data_location(self, table):
+        """
+        Retrieves the table data location s3 using glue meta store
```
Comment (datascienceChris) [inline hip_data_tools/aws/athena.py]:
```suggestion
        Retrieves the table's S3 data location using glue meta store
```

File: hip_data_tools/aws/athena.py [hunk 4]
```diff
@@ -212,6 +216,24 @@ def get_table_ddl(self, table):
             ddl = ddl + "\n"
         return ddl
 
+    def get_table_data_location(self, table):
```
Comment (datascienceChris) [inline hip_data_tools/aws/athena.py]: Do you want to write the type which is returned here?

File: hip_data_tools/aws/athena.py [hunk 5]
```diff
@@ -212,6 +216,24 @@ def get_table_ddl(self, table):
...
+            table (str): name of the table
+        Returns: a tuple of s3 bucket and key
+        """
+        table = self._get_glue_table_metadata(table)
+        location = table['Table']['StorageDescriptor']['Location']
+        bucket = location.split("/")[2]
+        key = "/".join(location.split("/")[3:])
+        return (bucket, key)
+
+    def _get_glue_table_metadata(self, table):
```
Comment (datascienceChris) [inline hip_data_tools/aws/athena.py]: there's a fair bit of logic going on here. Should this be a private method? You might wan to refactor

File: hip_data_tools/aws/s3.py [hunk 6]
```diff
@@ -133,14 +133,18 @@ def move_recursive_to_different_bucket(self, source_dir, destination_bucket_name
         if delete_after_copy:
             self.delete_recursive(source_dir)
 
+    def _resource(self):
```
Comment (datascienceChris) [inline hip_data_tools/aws/s3.py]: is this a getter method? get_resource?

File: hip_data_tools/aws/athena.py (hunk 1 above)
Comment (arihantsurana) [reply hip_data_tools/aws/athena.py]: inline

Comment (arihantsurana) [reply hip_data_tools/aws/athena.py]: also, I don't touch them ever.

File: hip_data_tools/aws/athena.py (hunk 5 above)
Comment (arihantsurana) [reply hip_data_tools/aws/athena.py]: changed to a single object creation and method call.

File: hip_data_tools/aws/s3.py (hunk 6 above)
Comment (arihantsurana) [reply hip_data_tools/aws/s3.py]: fixd

**Pull request merged**